kratos/geometries/triangle_2d_3.h

```diff
@@ -704,6 +704,43 @@ template<class TPointType> class Triangle2D3
         return false;
     }
 
+    /**
+     * Returns the local coordinates of a given arbitrary point
+     * The local coordinates are computed solving for xi and eta the system
+     * given by the linear transformation:
+     * x = (1-xi-eta)*x1 + xi*x2 + eta*x3
+     * y = (1-xi-eta)*y1 + xi*y2 + eta*y3
+     * @param rResult: The vector containing the local coordinates of the point
+     * @param rPoint: The point in global coordinates
+     * @return The vector containing the local coordinates of the point
+     */
+    CoordinatesArrayType& PointLocalCoordinates(
+        CoordinatesArrayType& rResult,
+        const CoordinatesArrayType& rPoint
+        ) override {
+
+        rResult = ZeroVector(3);
+
+        // Compute the Jacobian matrix and its determinant
+        bounded_matrix<double, 2, 2> J;
+        J(0,0) = this->GetPoint(1).X() - this->GetPoint(0).X();
+        J(0,1) = this->GetPoint(2).X() - this->GetPoint(0).X();
+        J(1,0) = this->GetPoint(1).Y() - this->GetPoint(0).Y();
+        J(1,1) = this->GetPoint(2).Y() - this->GetPoint(0).Y();
+        const double det_J = J(0,0)*J(1,1) - J(0,1)*J(1,0);
+
+        // Compute eta and xi
+        const double eta = (J(1,0)*(this->GetPoint(0).X()-rPoint(0)) +
+                            J(0,0)*(rPoint(1)-this->GetPoint(0).Y())) / det_J;
+        const double xi  = (rPoint(0) - this->GetPoint(0).X() - eta*J(0,1)) / J(0,0);
+
```

Comment:
Can we check if setting the number of iterations in one we get same as in geometry.h?, otherwise will mean that a change is needed